catalyst-gateway/bin/src/cardano/mod.rs

```diff
@@ -423,7 +423,11 @@ impl SyncTask {
                     } else if let Some(result) = finished.result.as_ref() {
                         match result {
                             Ok(()) => {
-                                self.current_sync_tasks -= 1;
+                                self.current_sync_tasks =
+                                    self.current_sync_tasks.checked_sub(1).unwrap_or_else(|| {
+                                        error!("current_sync_tasks -= 1 overflow");
```

Comment:
should it be an `error` or just a `warning` ?

Reply:
I'm not sure. I think it is indeed an error on our side indicating that we are doing the current task calculation wrong. I would even add a debug assertion here, so we can potentially catch this issue in the tests, but don't interfere with the normal (release) workflow.

Reply:
We never run anything in a `debug` build so the assertion would never do anything.

Reply:
It should log as an error if it happens (it should not),  But it should not actually fail anything, just saturate at 0.